lib/phase/flatten.js

```diff
@@ -306,6 +306,46 @@ function flattenSharedIntoJobs(shared, jobs) {
     return newJobs;
 }
 
+/**
+ * Merge shared steps into job when mergeSharedSteps annotation is set to true
+ * @param  {Object} sharedConfig               Shared config
+ * @param  {Object} jobConfig                  Job config
+ * @param  {Object} [template]                 Template
+ * @return {Object}                            Merged job config
+ */
+function handleMergeSharedStepsAnnotation({ sharedConfig, jobConfig, template }) {
+    const sharedSteps = sharedConfig.steps;
+    const newJobConfig = jobConfig;
+
+    // merge shared steps into oldJob
+    if (
+        sharedSteps &&
+        ((jobConfig.annotations && jobConfig.annotations['screwdriver.cd/mergeSharedSteps']) ||
```

Comment:
should we also check that jobConfig is using a template? I think that is one of the conditions to perform merging.